sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_bcl/HttpWebRequestFactory.cs

```diff
@@ -474,6 +474,14 @@ public virtual void ConfigureRequest(IRequestContext requestContext)
                 requestContext.Metrics.AddProperty(Metric.ProxyPort, requestContext.ClientConfig.ProxyPort);
                 _request.Proxy = proxy;
             }
+            else if (_request.RequestUri.Scheme == Uri.UriSchemeHttp)
+            {
+                _request.Proxy = requestContext.ClientConfig.GetHttpProxy();
+            }
+            else if (_request.RequestUri.Scheme == Uri.UriSchemeHttps)
```

Comment:
What is the point of putting in the code for HTTPS proxy with .NET Framework when .NET Framework doesn't support it. Users would get errors `The ServicePointManager does not support proxies with the https scheme.'`.

What I'm really worried is existing users running on machines that happened to have `HTTPS_PROXY` env var set and were being successfully ignored by their .NET Framework applications now start failing because we now attempt using `HTTPS_PROXY` on a platform that doesn't support it.

I think for .NET Framework we should only look for the `HTTP_PROXY` environment variable. But then that makes this kind of pointless since it is highly unlikely we would be going against an HTTP endpoint. Should we support the `ALL_PROXY` environment variable that would all using env var to configure an HTTP proxy for an HTTPS request?

Reply:
To be clear -- `HTTPS_PROXY` the environment variable doesn't have to connect to a proxy being served over HTTPS (what dotNET doesn't support until the release that just happened). Because while the .NET framework doesn't support connecting to a proxy being served over HTTPS, you can have a environment variable like: `export https_proxy="http://user:pass@localhost:8556/proxy"`, which uses HTTP to then establish an HTTPS tunnel through the proxy which is fully supported in all versions of dotnet (in fact it's what I'm trying to do!) I do also want to mention in dotnet 8 and later you should be able to connect to a proxy being served over HTTPS.

I think if users were setting `HTTPS_PROXY`, and it wasn't going through the proxy that'd be more unexpected, again this is what other SDKs do, it's not like we're adding unexpected behavior here. It's not really a variable people just set randomly in my experience? And again it does still have a chance of working! It'll work in all cases if they're using dotNET 8, or later.

I'm against adding support for ALL_PROXY as that is not widely supported by the other SDKs like `http_proxy`, and `https_proxy` are. I don't want to have to juggle yet another environment variable that no one else is using.

Reply:
That is a good point I missed that HTTPS_PROXY doesn't have to point to a proxy using HTTPS.